### PR TITLE
Handle empty address FormData in AddressWIthAutofillReviewField

### PR DESCRIPTION
**Pull request by coope93**
## Summary

- Handle loading `AddressWIthAutofillReviewField` when no state value is provided in the `formData` prop. 
- It was previously throwing an error because the result was `undefined` when trying to look up the state label, but now it will just render it as empty. 
- Added a test to validate when we pass an empty address and did a small refactor of the test file to DRY it up a bit. 

## Related issue(s)

https://github.com/department-of-veterans-affairs/va.gov-team/issues/83332

## Testing done
- Added FE test to handle this scenario

## What areas of the site does it impact?
Form 10-10 EZCG

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed
- [x] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [x] Browser console contains no warnings or errors.
- [x] Events are being sent to the appropriate logging solution

### Authentication

- [x] Did you login to a local build and verify all authenticated routes work as expected with a test user

### CODE REVIEW

File: src/applications/caregivers/tests/components/FormReview/AddressWithAutofillReviewField.unit.spec.js [hunk 1]
```diff
@@ -30,9 +18,26 @@ describe('CG <AddressWithAutofillReviewField>', () => {
...
+      street2: 'Suite 550',
+      city: 'Washington',
+      state: 'DC',
+      postalCode: '20005',
+      'view:autofill': false,
+    },
+    inputLabel: 'Primary Family Caregiver\u2019s',
+  };
+
+  const subject = (props = defaultProps) => {
```
Comment (coope93) [inline src/applications/caregivers/tests/components/FormReview/AddressWithAutofillReviewField.unit.spec.js:34]: I wrapped our `render` and `getSelectors` calls into a subject and set the `defaultProps` on it with the ability to pass in custom props.

File: src/applications/caregivers/tests/components/FormReview/AddressWithAutofillReviewField.unit.spec.js [hunk 2]
```diff
@@ -5,18 +5,6 @@ import { expect } from 'chai';
...
-    formData: {
-      street: '1350 I St. NW',
-      street2: 'Suite 550',
-      city: 'Washington',
-      state: 'DC',
-      postalCode: '20005',
-      'view:autofill': false,
-    },
-    inputLabel: 'Primary Family Caregiver\u2019s',
-  };
```
Comment (Mitch-A6) [inline src/applications/caregivers/tests/components/FormReview/AddressWithAutofillReviewField.unit.spec.js:19]: 🤔 Is shifting this below `getSelectors` for some visual locality or something? AFAICT the content's the same, but I don't get the reordering....

Comment (coope93) [reply src/applications/caregivers/tests/components/FormReview/AddressWithAutofillReviewField.unit.spec.js:19]: Yeah I just wanted it right above the subject so it was clearest where the defaultProps were being set.

**Review by Mitch-A6 — APPROVED** [review]
👍 Looks good!

**Review by harshilmoradia — APPROVED** [review]

**Review by mkerns1 — APPROVED** [review]
LGTM!

**Pull request merged**